src/libraries/Common/src/System/TimeProvider.cs

```diff
@@ -174,12 +198,19 @@ public override ITimer CreateTimer(TimerCallback callback, object? state, TimeSp
             /// </remarks>
             private sealed class SystemTimeProviderTimer : ITimer
             {
+#if SYSTEM_PRIVATE_CORELIB
                 private readonly TimerQueueTimer _timer;
-
+#else
+                private readonly Timer _timer;
+#endif // SYSTEM_PRIVATE_CORELIB
                 public SystemTimeProviderTimer(TimeSpan dueTime, TimeSpan period, TimerCallback callback, object? state)
                 {
                     (uint duration, uint periodTime) = CheckAndGetValues(dueTime, period);
+#if SYSTEM_PRIVATE_CORELIB
                     _timer = new TimerQueueTimer(callback, state, duration, periodTime, flowExecutionContext: true);
+#else
+                    _timer = new Timer(callback, state, duration, periodTime);
```

Comment:
CreateTimer needs to create a timer that's rooted as long as it's scheduled.  This isn't.  There are a variety of ways you can address that, such as using the Timer ctor that just takes a callback and subsequently calling Change on it; you'd then need to close over the actual callback and state in the lambda you use.